sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_exporter.py

```diff
@@ -73,8 +103,36 @@ def export(
         :return: The result of the export.
         :rtype: ~opentelemetry.sdk.metrics.export.MetricExportResult
         """
-        # TODO
-        return MetricExportResult.SUCCESS
+        result = MetricExportResult.SUCCESS
+        base_monitoring_data_point = kwargs.get("base_monitoring_data_point")
+        if metrics_data is None or base_monitoring_data_point is None:
+            return result
+        data_points = _metric_to_quick_pulse_data_points(
+            metrics_data,
+            base_monitoring_data_point=base_monitoring_data_point,
+            documents=kwargs.get("documents"),
+        )
+
+        token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        try:
+            post_response = self._client.post(  # type: ignore
+                monitoring_data_points=data_points,
+                ikey=self._instrumentation_key,
+                x_ms_qps_transmission_time=_ticks_since_dot_net_epoch(),
+                cls=Response,
+            )
+            if not post_response:
+                result = MetricExportResult.FAILURE
+            header = post_response._response_headers.get("x-ms-qps-subscribed")  # pylint: disable=protected-access
+            if header != "true":
+                # We raise an exception to indicate that quickpulse is not activated anymore
+                raise Exception()
```

Comment:
Does "not activated anymore" mean the customer left the live metrics page, right? is there a reason we need to raise an exception for this normal state? Is it caught somewhere?

Reply:
> Does "not activated anymore" mean the customer left the live metrics page, right? 

Yes that's what that means. I've changed this to return `MetricExportResult.FAILURE` since that is essentially an "unsuccessful" case instead of throwing an `Exception`.

Reply:
I also created a custom error `UnsuccessfulQuickPulsePostError` which is just simple inheritance of base class `Exception` that is thrown in `_receive_metrics` when `MetricExportResult.FAILURE` is returned from `export`. The custom exception is then handled in `_ticker` to appropriately handle the backoff logic.